### DISCUSSION TITLE
SQLite3 migration to mariaDB

### DISCUSSION THREAD

**Discussion by adriantalpa** [shlinkio/discussions#90] [Help wanted] [answered]
Hi,
is there a way to migrate from sqlite3 to mariaDB database?

I tried to port the data using sqlite3mysql but I get 
`MySQL failed adding foreign key to short_urls.domain_id referencing domains.id: 3780 (HY000): Referencing column 'domain_id' and referenced column 'id' in foreign key constraint 'short_urls_FK_1_0' are incompatible.
`  when I use `--ignore-duplicate-keys`. Anything else I tried doesn't work...

Please help. Any ideas?
Cheers

**Comment by acelaya** [comment, marked as answer]
You can try using the import tool (`shlink short-url:import`).

It allows you to import from different sources, including another Shlink instance.

You could setup a new instance with Mariadb, and import from the original instance. Then, you can get rid of the old one and replace it.

You can find more info here https://shlink.io/documentation/advanced/import-short-urls/#shlink-instance

**Comment by adriantalpa** [comment]
Thanks for the quick answer!
Looks like the solution was right under my nose in the doc pages: "_... or you want to change the database engine for..._ "

Adrian
